### PR TITLE
Add SBOM support

### PR DESCRIPTION
**Pull request by amanshallow**

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -30,6 +30,12 @@ builds:
 archives:
 - format: zip
   name_template: '{{ .ProjectName }}_{{ .Version }}_{{ .Os }}_{{ .Arch }}'
+sboms:
+- documents:
+  - bom.json
+  artifacts: any
+  cmd: cyclonedx-gomod
+  args: [ "mod", "-assert-licenses", "-std", "-json", "-verbose", "-output", "$document", "../" ]
```
Comment (gemcdaniel) [inline .goreleaser.yml:38]: This will need to be modified just like the others as it won't have all the information required in the generated SBOM.

Comment (amanshallow) [reply .goreleaser.yml:38]: Made it so it generates an SBOM for each binary. IIUC, it should have the right options set. I don't see a way we could merge the different SBOMs, however.

Comment (amanshallow) [reply .goreleaser.yml:38]: Tried testing it by pushing up a tag but it doesn't seem like GitHub will let me create a test release without everything being merged in `main`.

Comment (developingjames) [reply .goreleaser.yml:38]: We have another issue that is to flush out the sbom details.

**Review by developingjames — APPROVED** [review]

**Pull request closed without merging**